lib/serve/functions.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+var _ = require('lodash');
+var chalk = require('chalk');
+var path = require('path');
+var RSVP = require('rsvp');
+var getProjectId = require('../getProjectId');
+var logger = require('../logger');
+var utils = require('../utils');
+var parseTriggers = require('../parseTriggers');
+var EmulatorController;
+var controller;
+
+function _pollOperation(service, operation) {
```

Comment:
I did have to change `controller.getOperation(operation[0].name)` to `controller.**client.**getOperation(operation[0].name)`